pytket/extensions/cutensornet/tensor_network_convert.py

```diff
@@ -483,9 +483,65 @@ def vdot(self, tn_other: "TensorNetwork") -> list:
         return tn_concatenated
 
 
-class PauliOperatorTensorNetwork:
-    """Handles a tensor network representing a Pauli operator string."""
+def _measure_qubit_state(
+    ket: TensorNetwork, qubit_id: Qubit, bit_value: int, loglevel: int = logging.INFO
+) -> TensorNetwork:
+    """Measures a qubit in a tensor network. by appending a measurement
+    gate to the tensor network.
+    The measurment gate is applied via appending a tensor cap of
+    the form:  0: [1, 0] or 1: [0, 1] to the interleaved einsum input.
+    Therefor removing one of the open indices of the tensor network.
+
+    Args:
+        ket: a TensorNetwork object representing a quantum state.
+        qubit_id: a qubit id.
+        bit_value: a bit value to be assigned to the measured qubit.
+        loglevel: logging level.
+
+    Returns:
+        A TensorNetwork object representing a quantum state after the
+        measurement with a modified interleaved notation containing the extra
+        measurement tensor.
+    """
+
+    cap = {
+        0: np.array([1, 0], dtype="complex128"),
+        1: np.array([0, 1], dtype="complex128"),
+    }
+
+    sticky_ind = ket.sticky_indices[qubit_id]
+    ket._cuquantum_interleaved.extend([cap[bit_value], [sticky_ind]])
```

Comment:
It will not work because dagger uses the underlying circuit class which does not know about post selection. Hence it would be better to dagger the network itself not the circuit. I made an issue about this.

Reply:
True story... We need to think about it.

Reply:
But we need to remove the capped sticky indices anyway as it is a path to bugs...